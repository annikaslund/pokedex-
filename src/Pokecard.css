.Pokecard {
    background-color: grey;  
    display: inline-block;
}

h5 {
    color: blue;
}
