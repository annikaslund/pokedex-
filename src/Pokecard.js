import React, { Component } from 'react';
import "./Pokecard.css"
class Pokecard extends Component {
    render(){
        let { name, id, type, base_experience } = this.props;
        let imgURL = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`
        return (
            <div className="Pokecard">
                <h5>{ name }</h5>
                <img src={imgURL} className="pokemon-image" alt="pokemon"/>
                <p>Type: { type }</p>
                <p>EXP: { base_experience }</p>
            </div>
        );
    }
}

export default Pokecard;