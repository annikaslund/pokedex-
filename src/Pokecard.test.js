import React from 'react';
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Pokecard from './Pokecard';

//smoke test
it('renders without crashing', function() {
  shallow(<Pokecard />);
});

//snapshot test
it('matches snapshot', function() {
  let wrapper = shallow(<Pokecard />); //taking snapshot as saved snapshot
  let serialized = toJson(wrapper); //rendering snapshot
  expect(serialized).toMatchSnapshot();
});