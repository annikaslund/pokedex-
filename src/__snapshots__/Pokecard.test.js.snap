// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Pokecard"
>
  <h5 />
  <img
    alt="pokemon"
    className="pokemon-image"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
  />
  <p>
    Type: 
  </p>
  <p>
    EXP: 
  </p>
</div>
`;
